MM-54814: Convert ./components/widgets/menu/menu_group.tsx from Class Component to Function Component

KuSh opened the pull request:
<!-- Thank you for contributing a pull request! Here are a few tips to help you:

1. If this is your first contribution, make sure you've read the Contribution Checklist https://developers.mattermost.com/contribute/getting-started/contribution-checklist/
2. Read our blog post about "Submitting Great PRs" https://developers.mattermost.com/blog/2019-01-24-submitting-great-prs
3. Take a look at other repository specific documentation at https://developers.mattermost.com/contribute
-->

#### Summary
<!--
A description of what this pull request does, as well as QA test steps (if applicable and if not already added to the Jira ticket).
-->
Convert ./components/widgets/menu/menu_group.tsx from Class Component to Function Component

#### Ticket Link
<!--
If applicable, please include both or either of the following links:
-->

Fixes https://github.com/mattermost/mattermost/issues/24775
Jira https://mattermost.atlassian.net/browse/MM-54814

#### Screenshots
<!--
If the PR includes UI changes, include screenshots/GIFs.

For an easier comparison of UI changes a table (template below) can be used.

|  before  |  after  |
|----|----|
| <insert before screenshot here> | <insert after screenshot here> |

-->

#### Release Note
<!--
Add a release note for each of the following conditions:

* Config changes (additions, deletions, updates).
* API additions—new endpoint, new response fields, or newly accepted request parameters.
* Database changes (any).
* Schema migration changes. Use the [Schema Migration Template](https://docs.google.com/document/d/18lD7N32oyMtYjFrJKwsNv8yn6Fe5QtF-eMm8nn0O8tk/edit?usp=sharing) as a starting point to capture these details as release notes. 
* Websocket additions or changes.
* Anything noteworthy to a Mattermost instance administrator (err on the side of over-communicating).
* New features and improvements, including behavioral changes, UI changes, and CLI changes.
* Bug fixes and fixes of previous known issues.
* Deprecation warnings, breaking changes, or compatibility notes.

If no release notes are required, write NONE. Use past-tense. Newlines are stripped.

Examples:

```
Added new API endpoints POST /api/v4/foo, GET api/v4/foo, and GET api/v4/foo/:foo_id.
```

```
Added a new config setting ServiceSettings.FooBar. Added a new column Foo to the Users table.
```

```
NONE
```
-->
```release-note
NONE
```

mattermost-build commented:
Hello @KuSh,

Thanks for your pull request! A Core Committer will review your pull request soon. For code contributions, you can learn more about the review process [here](https://developers.mattermost.com/contribute/getting-started/code-review/).

larkox commented on webapp/channels/src/components/widgets/menu/menu_group.tsx:
```diff
@@ -8,32 +8,30 @@ import './menu_group.scss';
-        );
-    }
-}
+export default MenuGroup;
```
The component used to be a pure component, so add here `React.memo`.

KuSh replied:
Oh nice trick ! Sure done

larkox reviewed (changes requested):
Great work! Just one small thing.

mattermost-build commented:
E2E tests not automatically triggered, because the PR is not in a mergeable state. Please update the branch with the base branch and resolve outstanding conflicts.

sinansonmez reviewed (approved):
LGTM, thanks for the effort

larkox commented:
@KuSh There are failing tests. Most of them (if not all) can be fixed by running `npm run test -- -u` in the channels folder. Would you mind fixing them?

KuSh commented:
> @KuSh There are failing tests. Most of them (if not all) can be fixed by running `npm run test -- -u` in the channels folder. Would you mind fixing them?

Sure !

KuSh commented:
I've fixed tests and updated snapshots @larkox

larkox commented:
Sorry for the back and forth @KuSh . There seems to be now some conflict with master. Can you merge master into your branch and fix the conflict?

KuSh commented:
> Sorry for the back and forth @KuSh . There seems to be now some conflict with master. Can you merge master into your branch and fix the conflict?

No problem! I've rebased and fixed the conflict @larkox

mattermost-build commented:
This PR has been automatically labelled "stale" because it hasn't had recent activity.
A core team member will check in on the status of the PR to help with questions.
Thank you for your contribution!

larkox commented:
@KuSh There seems to be another merge conflict. Really sorry for that :sweat:

KuSh commented:
Done @larkox 👍

larkox reviewed (approved):
Thank you very much!

lindy65 reviewed (approved):
Thanks @KuSh looks good to me 👍 

cc @larkox
